test/test_data/test_validation.py

```diff
@@ -229,3 +294,234 @@ def test_dataframe_econding_2D(self):
+        y[1] = np.nan
+        y = pd.DataFrame(y)
+
+        with self.assertRaisesRegex(ValueError, 'Categorical features array cannot contain'):
+            InputValidator().validate_features(x)
+
+        with self.assertRaisesRegex(ValueError, 'Target values cannot contain missing/NaN'):
+            InputValidator().validate_target(y, is_classification=True)
+
+        with self.assertRaisesRegex(ValueError, 'Target values cannot contain missing/NaN'):
+            InputValidator().validate_target(y, is_classification=False)
+        return
+
+    def test_no_new_category_after_fit(self):
+        # First make sure no problem if no categorical
+        x = pd.DataFrame({'A': [1, 2, 3, 4], 'B': [5, 6, 7, 8]})
+        y = pd.DataFrame([1, 2, 3, 4])
+        validator = InputValidator()
+        validator.validate(x, y, is_classification=True)
+        validator.validate_features(x)
+        x['A'] = x['A'].apply(lambda x: x*x)
+        validator.validate_features(x)
+
+        # Then make sure we catch categorical extra categories
+        x = pd.DataFrame({'A': [1, 2, 3, 4], 'B': [5, 6, 7, 8]}, dtype='category')
+        y = pd.DataFrame([1, 2, 3, 4])
+        validator = InputValidator()
+        validator.validate(x, y, is_classification=True)
+        validator.validate_features(x)
+        x['A'] = x['A'].apply(lambda x: x*x)
+        with self.assertRaisesRegex(
+            ValueError,
+            'During fit, the input features contained categorical values'
+        ):
+            validator.validate_features(x)
+
+        # For label encoder of targets
+        with self.assertRaisesRegex(
+            ValueError,
+            'During fit, the target array contained the categorical'
+        ):
+            validator.validate_target(pd.DataFrame([1, 2, 5, 4]))
+
+        # For ordinal encoder of targets
+        x = pd.DataFrame({'A': [1, 2, 3, 4], 'B': [5, 6, 7, 8]}, dtype='category')
+        validator = InputValidator()
+        validator.validate(x, x, is_classification=True)
+        validator.validate_target(pd.DataFrame(
+            {'A': [1, 2, 3, 4], 'B': [5, 6, 7, 8]}, dtype='category')
+        )
+        with self.assertRaisesRegex(
+            ValueError,
+            'During fit, the target array contained the categorical'
+        ):
+            validator.validate_target(pd.DataFrame(
+                {'A': [1, 2, 3, 4], 'B': [5, 9, 7, 8]}, dtype='category')
+            )
+        return
+
+    def test_big_dataset_encoding(self):
+        x, y = sklearn.datasets.fetch_openml(data_id=2, return_X_y=True, as_frame=True)
+        validator = InputValidator()
+
+        with self.assertRaisesRegex(
+            ValueError,
+            'Categorical features array cannot contain missing/NaN values'
+        ):
+            x_t, y_t = validator.validate(x, y, is_classification=True)
+
+        # Make sure translation works apart from Nan
+
+        # Drop columns that are all nan
+        x = x.dropna('columns')
```

Comment:
Is it possible to not drop all-nan columns outside the scikit-learn pipeline?

Reply:
IYes, it is. Added a code for this